### ISSUE TITLE
AST configuration options

### ISSUE THREAD

**Issue by veggiesaurus** [idia-astro/carta-frontend#38]
AST config. options:
- if it is RA/DEC, use sexagesimal format by default, rather than in degree format.
		(e.g., R.A. = 12h:23m:43.234s, Dec = -34d:35m:23.123s)
- if it is galactic coordinate, use degree format by default
		(e.g., GLON = 23.123 degree, GLAT = -5.234 degree)
- "WCS(a, b)" in the cursor info bar should follow the above two rules
- default format for "Numbers" as "d.3"

- Overlay settings -> Global -> Color (not functional; this is supposed to change the color of ticks, grid, label, etc. in one shot?)

**Comment by arusstaylor** [comment]
d.3 is no likely to be enough precision for modern images.  Typical JVLA, GMRT and MeerKAT pixels sizes are of order 1 arc or less, which is a ~10^-4 degrees.  Is it possible to link the precision of the display format to the pixel size of the image?  For a VLBI images, for example, the pixels may be milli-arcseconds, which is ~10^-7 degrees.  I think it would be good if the precision of the display were d.3 as minimum but is adapted to the image pixel size such  that the last digit of the display coordinate will change from pixel to pixel.

Russ

**Comment by veggiesaurus** [comment]
As @kswang1029 was the author of the initial issue (I just copied it), hopefully he will comment on that.
 
@arusstaylor What I'd think would be ideal would be to determine the number of decimal places required based on the current image view. So for example, if your view is initially quite wide, 3 decimals might be sufficient, but as you zoom in, it should include more decimal places. That avoids the issue of having a whole number of decimal places that aren't useful, because the field of view is wide.

**Comment by kswang1029** [comment]
@arusstaylor @veggiesaurus Right, modern images can have quite small pixel sizes so that d.3 is not sufficient. As @veggiesaurus suggested, I think what we display is determined by two factors, one is the FOV, which determines the minimum precision, the other is the pixel size, which then determines the maximum precision. As we zoom in, we increase the precision dynamically. 
If this may take some time to implement, perhaps we can set "d.3" as default just for the release. Currently "d.1" is not enough in general cases.

**Comment by arusstaylor** [comment]
If we have to go with a fixed value default for the initial release then I recommend d.5 so we have arc second precision.  Most of the images from MeerKAT will require this.

Russ

**Comment by kswang1029** [comment]
@arusstaylor Sure for d.5 as a temporary fixed default.

**Comment by veggiesaurus** [comment]
@kswang1029 done. I don't think the dynamic precision is too difficult though. We have all the information we need. Perhaps we could actually simplify things and use the pixel size of the downsampled image to guide our choice?

**Comment by arusstaylor** [comment]
Sounds reasonable.
Russ

**Comment by confluence** [comment]
Update: it looks like setting a specific number of decimal places in the format (C) also affects the formatting of the cursor coordinate values, but setting the total number of digits (B) does not. By default, if the number of decimal places is not explicitly set, AST uses 4 decimal places for cursor values in degrees and one decimal place for sexagesimal cursor values. Is this going to be a problem? Do we want to be able to modify these defaults?

**Comment by arusstaylor** [comment]
Yes, we will want to override these defaults.   This is not enough for images with angular resolution better than a few arc seconds.

Russ

**Comment by confluence** [comment]
OK. For now I am restoring the ability to set the number of decimal places, since that affects both the axes and the cursor. I have not yet found a way to control them separately -- they appear to use the same setting.

**Comment by confluence** [comment]
Update: I've worked out how we control the cursor format, and I can add a separate setting for it if that is desirable. Do we want that, or is a single precision setting for the axes and the cursor sufficient?

**Comment by kswang1029** [comment]
> @kswang1029 done. I don't think the dynamic precision is too difficult though. We have all the information we need. Perhaps we could actually simplify things and use the pixel size of the downsampled image to guide our choice?

This is a good idea. At high zoom level (~per pixel scale), we will need the precision at sub-pixel scale (e.g., 1/10th of pixel size).

**Comment by kswang1029** [comment]
> Update: I've worked out how we control the cursor format, and I can add a separate setting for it if that is desirable. Do we want that, or is a single precision setting for the axes and the cursor sufficient?

I think we will need a separate setting for each. For axes, we will just need to precision sufficient to distinguish each tick label. For the cursor, we will need higher precision at sub-pixel scale.

**Comment by confluence** [comment]
I have added a separate precision setting for the cursor (currently the format is the same as the format used on the axes).

Still left to do: setting a default format based on the coordinate system of the image.

**Comment by arusstaylor** [comment]
I am not sure the need a user option for the precision of the labelling for the coordinate grid.  This should just automatically label with sufficient precision that the number are different on adjacent ticks.  

Russ

**Comment by confluence** [comment]
That is the current default behaviour, with the option to override the precision. I can remove the option entirely, if we don't need it.

**Comment by arusstaylor** [comment]
Fine to have the option as long as the default behaviour is correct.

Russ

**Comment by confluence** [comment]
To clarify: for RA/DEC, would you like `hms` for RA and `dms` for DEC (*not* `hms` for both)? And just degrees for GLON *and* GLAT?

**Comment by kswang1029** [comment]
Yes, hms for R.A. and dms for Dec. GLON and GLAT are both in degrees.

**Comment by confluence** [comment]
To clarify: for RA/DEC, would you like `hms` for RA and `dms` for DEC (*not* `hms` for both)? And just degrees for GLON *and* GLAT? This is the default behaviour that I have now added.

**Comment by arusstaylor** [comment]
Yes.
Russ
